detect_secrets/main.py

```diff
@@ -73,6 +74,11 @@ def main(argv=None):
                 )
 
     elif args.action == 'audit':
+        if args.report:
+            report.execute(args)
+
+        report.validate_args(args, parserBuilder.subparser.choices['audit'])
```

Comment:
We need this check to account for when reporting arguments are included without `--report`.